manifest: update trusted-firmware-m to latest upstream

ioannisg opened the pull request:
Synchronize TF-M to the latest ustream main branch.
This brings TF-M support for BL5340 Dev Kit in Zephyr.

The PR fixes also a minor issue in the psa_level1 sample.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
manifest: update trusted-firmware-m to latest upstream

Synchronize TF-M to the latest ustream main branch.
This brings TF-M support for BL5340 Dev Kit in Zephyr.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
FYI @greg-leach

ioannisg committed:
samples: psa_leve1: do not re-initialize NS interface

TF-M NS interface is initialized by the TF-M module
code, so it does not need to be re-initialized in the
sample code.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

microbuilder reviewed (approved):
Tested with `mps2_an521_nonsecure` and `lpcxpresso55s69_ns`.

microbuilder commented:
I don't really love not being on a fixed release for certification reasons, since it makes the certificate reference a mess, but it does pull in some other changes that are useful before 1.4.0 is tagged upstream (which will be post Zephyr 2.6.0).

ioannisg commented:
> I don't really love not being on a fixed release for certification reasons, since it makes the certificate reference a mess, but it does pull in some other changes that are useful before 1.4.0 is tagged upstream (which will be post Zephyr 2.6.0).

Let's hold this off, until we know we will be able to merge the support for the BL5340 DK. Otherwise, this could be done post v2.6.0.

ioannisg commented:
> Tested with `mps2_an521_nonsecure` and `lpcxpresso55s69_ns`.

Thanks. Myself, tested this on nrf5340dk, nrf9160dk and nucleo_l552ze_q_ns

ioannisg committed:
tests: arm_thread_swap_tz: add tfm tag

Add the tfm tag to the arm_thread_swap_tz test,
since the test is running with TF-M by default.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>
